front/lib/swr/mcp_servers.ts

```diff
@@ -194,6 +199,34 @@ export function useCreateInternalMCPServer(owner: LightWorkspaceType) {
   return { createInternalMCPServer };
 }
 
+/**
+ * Hook to check if an OAuth connection is required for a remote MCP server.
+ */
+export function useCheckOAuthConnection(owner: LightWorkspaceType) {
```

Comment:
I think we should name it useRetrieveOAuthParameters because it not only checks but also creates a client secret/id pair